desktop_version/CMakeLists.txt

```diff
@@ -158,38 +158,32 @@ ELSE()
 	ADD_EXECUTABLE(VVVVVV ${VVV_SRC})
 ENDIF()
 
-IF(NOT OFFICIAL_BUILD)
-	# Add interim commit hash and its date to the build
-
-	# FIND_PACKAGE sets GIT_FOUND and GIT_EXECUTABLE
-	FIND_PACKAGE(Git)
-
-	IF(GIT_FOUND)
-		# These filenames have to be qualified, because when we run
-		# the CMake script, its work dir gets set to the build folder
-		SET(VERSION_INPUT_FILE ${CMAKE_CURRENT_SOURCE_DIR}/src/Version.h.in)
-		SET(VERSION_OUTPUT_FILE ${CMAKE_CURRENT_SOURCE_DIR}/src/Version.h.out)
-
-		ADD_CUSTOM_TARGET(
-			GenerateVersion ALL
-			# This BYPRODUCTS line is required for this to be ran every time
-			BYPRODUCTS ${VERSION_OUTPUT_FILE}
-			COMMAND ${CMAKE_COMMAND}
-			# These args have to be passed through, otherwise the script can't see them
-			# Also, these args have to come BEFORE `-P`! (Otherwise it fails with an unclear error)
-			-DGIT_EXECUTABLE=${GIT_EXECUTABLE}
-			-DINPUT_FILE=${VERSION_INPUT_FILE}
-			-DOUTPUT_FILE=${VERSION_OUTPUT_FILE}
-			-P ${CMAKE_CURRENT_SOURCE_DIR}/version.cmake
-		)
-
-		ADD_DEPENDENCIES(VVVVVV GenerateVersion)
-
-		# This lets Version.h know that Version.h.out exists
-		ADD_DEFINITIONS(-DVERSION_H_OUT_EXISTS)
-	ENDIF()
+FIND_PACKAGE(Git REQUIRED)
+
+# These filenames have to be qualified, because when we run
+# the CMake script, its work dir gets set to the build folder
+SET(VERSION_INPUT_FILE ${CMAKE_CURRENT_SOURCE_DIR}/src/Version.h.in)
+SET(VERSION_OUTPUT_FILE ${CMAKE_CURRENT_SOURCE_DIR}/src/Version.h)
+
+ADD_CUSTOM_TARGET(
+	GenerateVersion ALL
+	# This BYPRODUCTS line is required for this to be ran every time
+	BYPRODUCTS ${VERSION_OUTPUT_FILE}
+	COMMAND ${CMAKE_COMMAND}
+	# These args have to be passed through, otherwise the script can't see them
+	# Also, these args have to come BEFORE `-P`! (Otherwise it fails with an unclear error)
+	-DGIT_EXECUTABLE=${GIT_EXECUTABLE}
+	-DINPUT_FILE=${VERSION_INPUT_FILE}
+	-DOUTPUT_FILE=${VERSION_OUTPUT_FILE}
+	-P ${CMAKE_CURRENT_SOURCE_DIR}/version.cmake
+)
+
+IF(OFFICIAL_BUILD)
+	ADD_DEFINITIONS(-DOFFICIAL_BUILD)
```

Comment:
You completely misunderstand the `OFFICIAL_BUILD` CMake variable - it's not a compile-time definition. Nowhere in the code is there an `#ifdef OFFICIAL_BUILD`.

Reply:
The PR introduces one to decide whether or not to include the Commit Date; see Version.h.in

Reply:
My bad then, I didn't see that.